slack.ts

```diff
@@ -0,0 +1,24 @@
+/** Error20:
+    - line 21: don't know how to declare call signature in tsc*/
+
+import * as querystring from "querystring";
+import * as fetch from "node-fetch";
+
+const tkn = process.env.SLACK_TOKEN;
+
+export async function slackChat(slackDomain : string, slackChannel : string, slackMessage : string) {
```

Comment:
So I think this is getting really good, now you just need to change it a little to a format the server expects.
So far I think every thing will have to implement:
```typescript
    async function send(message: string, config: T): Promise<APIReturn> {}
```

So all that means is make a function like that and put all the other arguments (`channel` in this case) in one object that you pass in as `config`. I think since the domain is tied to the slack token, you can just get it like your getting the token `process.env`.

For now you can just return an error report of type: `{ error: boolean; message?: string;}`

Reply:
By this do you want me to make a new function called send or to edit the heading of my existing function so it looks like this?

Reply:
editing the heading is good